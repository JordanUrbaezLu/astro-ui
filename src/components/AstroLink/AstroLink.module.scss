@use "../../utility/globals";

.astroLinkContainer {
  font-family: Verdana, Geneva, Tahoma, sans-serif;
  position: relative;
  text-decoration: none;

  &.primary {
    color: globals.$primary-color;
  }

  &.primary::after {
    content: "";
    position: absolute;
    width: 100%;
    transform: scaleX(0);
    height: 1.5px;
    bottom: 0;
    left: 0;
    transform-origin: bottom right;
    background-color: globals.$primary-color;
    transition: transform 0.25s ease-out;
  }

  &.secondary {
    color: globals.$secondary-color;
  }

  &.secondary::after {
    content: "";
    position: absolute;
    width: 100%;
    transform: scaleX(0);
    height: 1.5px;
    bottom: 0;
    left: 0;
    transform-origin: bottom right;
    background-color: globals.$secondary-color;
    transition: transform 0.25s ease-out;
  }

  &:hover::after {
    transform: scaleX(1);
    transform-origin: bottom left;
  }
}
