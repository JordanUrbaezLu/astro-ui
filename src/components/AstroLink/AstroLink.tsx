import classNames from "classnames";
import * as React from "react";
import styles from "./AstroLink.module.scss";

export type AstroLinkVariant = "primary" | "secondary";

export interface AstroLinkProps {
  /**
   * The content for the AstroLink
   */
  children: string;
  /**
   * The href for the AstroLink
   */
  href: string;
  /**
   * The variant for the AstroLink
   *
   * @default "primary"
   */
  variant?: AstroLinkVariant;
}

export const AstroLink: React.FC<AstroLinkProps> = (props) => {
  const { children, href, variant = "primary" } = props;

  const linkContainer = classNames(
    styles.astroLinkContainer,
    variant === "primary" ? styles.primary : styles.secondary
  );

  return (
    <a className={linkContainer} href={href}>
      {children}
    </a>
  );
};
