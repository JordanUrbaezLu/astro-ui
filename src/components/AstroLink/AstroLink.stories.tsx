import { Story } from "@storybook/react";
import { AstroLink, AstroLinkProps } from "./AstroLink";

export default {
  component: AstroLink,
  title: "Components/AstroLink",
};

const Template: Story<Omit<AstroLinkProps, "children" | "href">> = (args) => {
  return (
    <AstroLink children="Google.com" href="https://www.google.com/" {...args} />
  );
};

export const Default = Template.bind({});

export const Primary = Template.bind({});
Primary.args = {
  variant: "primary",
};

export const Secondary = Template.bind({});
Secondary.args = {
  variant: "secondary",
};
