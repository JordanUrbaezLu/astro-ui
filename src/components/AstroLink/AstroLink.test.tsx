import { render, screen } from "@testing-library/react";
import AstroLink, { AstroLinkVariant } from "./AstroLink";
import { axe } from "jest-axe";

describe("AstroLink", () => {
  test("Should render component correctly.", () => {
    const { container } = render(
      <AstroLink href="https://www.google.com/">Google.com</AstroLink>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe("Accessibility", () => {
  test("Should have no accessibility violations.", async () => {
    const { container } = render(
      <AstroLink href="https://www.google.com/">Google.com</AstroLink>
    );
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});

describe("Props", () => {
  test.each<AstroLinkVariant>(["primary", "secondary"])(
    "Should render variants correctly.",
    (variant) => {
      render(
        <AstroLink href="https://www.google.com/" variant={variant}>
          Google.com
        </AstroLink>
      );
      expect(screen.getAllByRole("link")[0]).toHaveClass(variant);
    }
  );
});
