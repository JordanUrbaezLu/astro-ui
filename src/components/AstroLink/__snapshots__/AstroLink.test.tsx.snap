// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AstroLink Should render component correctly. 1`] = `
<a
  class="astroLinkContainer primary"
  href="https://www.google.com/"
>
  Google.com
</a>
`;
